subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -778,12 +806,28 @@ int hawkbit_default_config_data_cb(const char *device_id, uint8_t *buffer, const
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
 int hawkbit_set_config(struct hawkbit_runtime_config *config)
 {
+	size_t length;
+
 	if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
 		if (config->server_addr != NULL) {
+			hb_cfg.server_addr_set = true;
 			strncpy(hb_cfg.server_addr, config->server_addr,
 				sizeof(hb_cfg.server_addr));
 			LOG_DBG("configured %s: %s", "hawkbit/server_addr", hb_cfg.server_addr);
 		}
+		if (config->server_hostname != NULL) {
```

Comment:
That's an array so it can't be NULL, no?

Reply:
`config` is of type [`hawkbit_runtime_config`](https://github.com/zephyrproject-rtos/zephyr/pull/89374/files/c776b00c6a0e4ebf8e3b9165e785adba1df8fca7#diff-ad25768e7f4edf2a111d7eaaf97f7aef126d77d3b4852c332706df975cdf7e04L31), and `server_hostname` is a `char *`.

The `config` structure is for the user to provide the string values to the hawkbit internals, where they are copied into the internal `hb_cfg`.